lib/plug/conn.ex

```diff
@@ -542,8 +561,8 @@ defmodule Plug.Conn do
     raise AlreadySentError
   end
 
-  def put_req_header(%Conn{adapter: adapter, req_headers: headers} = conn, key, value) when
-      is_binary(key) and is_binary(value) do
+  def put_req_header(%Conn{adapter: adapter, req_headers: headers} = conn, key, value)
```

Comment:
Maybe?

```elixir
def put_req_header(%Conn{} = conn, key, value) when is_binary(key) and is_binary(value) do
  validate_header_key_if_test!(conn.adapter, key)
  %{conn | req_headers: List.keystore(conn.req_headers, key, 0, {key, value})}
end
```

Reply:
Doesn't change much, I prefer the way it is right now.